### PR TITLE
Add Autograd doc to gluon page

### LINKED ISSUE #7306: Gluon / Autograd is_train documentation
As far as I understand `AutogradRuntime::Get()->IsTraining()` is used throughout the [NDArray API](https://github.com/apache/incubator-mxnet/blob/dad83816511ec04157d2743ee1f220e74a6b0c27/src/c_api/c_api_ndarray.cc) to determine how to set the `OpContext` argument to the functions.

`with autograd.record()`sets the training attribute of AutogradRuntime to True and resets it when leaving the block.

Is this correct? This should be documented. The current Gluon documentation does not state anything about how `OpContext.is_train` is determined, even though this is very relevant to the BatchNorm and Dropout operators. If you confirm my understanding I can add a short note to the Gluon Tutorial. If a documentation page for autograd is added it would make sense to also mention it there.

### PR DESCRIPTION
**Pull request by leezu**
I can't build the docs locally so this is currently untested. Fixes #7306. In reply to https://github.com/apache/incubator-mxnet/pull/7303#issuecomment-319790394

### CODE REVIEW

**Comment by piiswrong** [comment]
autograd is under mxnet so should be under top level.

**Comment by leezu** [comment]
I moved the docs to a new toplevel file now
